src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/TransactionValidator.cs

```diff
@@ -5,35 +5,61 @@
 
 using System;
 using System.Collections.Generic;
-using Microsoft.Health.Fhir.Api.Features.Routing;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Primitives;
 using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Conformance;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Resources;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Messages.Search;
 using Microsoft.Health.Fhir.Core.Models;
 using static Hl7.Fhir.Model.Bundle;
 
 namespace Microsoft.Health.Fhir.Api.Features.Resources.Bundle
 {
-    public static class TransactionValidator
+    public class TransactionValidator : BaseConditionalHandler
```

Comment:
This _seems_ like its just validating resource references as in `BundleReferenceValidator`, thoughts?

Reply:
The validation is only reuired for the transaction bundle and therefore it was name TransactionValidator. I can rename it to TransactionBundleReferenceValidator maybe?

Reply:
This seems more descriptive to me, initially I thought it was validating that the transaction actions happened successfully

Reply:
I renamed it to TransactionBundleValidator  because in addition to validating bundlereference, it also validates uniques ogf entries even if the entry doesnot contain reference and also checks if there are entries with operations that are not allow. So that name seemed appropriate,